datafusion/expr/src/built_in_function.rs

```diff
@@ -464,12 +479,16 @@ impl BuiltinScalarFunction {
             },
             BuiltinScalarFunction::ArrayConcat => {
                 let mut expr_type = Null;
+                let mut max_dims = 1;
```

Comment:
I think max_dims should be 0, If all the list is 1D, we will get null for expr_type.